warehouse/keyword-common/src/main/java/datawave/util/keyword/language/YakeLanguage.java

```diff
@@ -0,0 +1,85 @@
+package datawave.util.keyword.language;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import com.ibm.icu.text.BreakIterator;
+
+/**
+ * Defines an extensible interface for language management, providing methods to get language specific stopwords, sentence break iterators and/or register new
+ * languages.
+ */
+public interface YakeLanguage {
+    String getLanguageCode();
+
+    Set<String> getStopwords();
+
+    BreakIterator getSentenceBreakIterator();
+
+    /** Manages an extendable collection of languages */
+    class Registry {
+        static final Map<String,YakeLanguage> languageRegistry = new HashMap<>();
+
+        public static void add(YakeLanguage language) {
+            languageRegistry.put(language.getLanguageCode(), language);
+        }
+
+        public static YakeLanguage find(String languageCode) {
+            return languageRegistry.get(languageCode);
+        }
+    }
```

Comment:
any concern about thread safety?

Reply:
Not really, I'll document the expected usage here in javadoc that should explain why we might not need to be as concerned about thread safety here.